pkg/scheduler/framework/plugins/podtopologyspread/filtering_test.go

```diff
@@ -3380,8 +3380,8 @@ func TestPreFilterDisabled(t *testing.T) {
 	cycleState := framework.NewCycleState()
 	gotStatus := p.(*PodTopologySpread).Filter(context.Background(), cycleState, pod, nodeInfo)
 	wantStatus := framework.AsStatus(fmt.Errorf(`reading "PreFilterPodTopologySpread" from cycleState: %w`, framework.ErrNotFound))
-	if !reflect.DeepEqual(gotStatus, wantStatus) {
-		t.Errorf("status does not match: %v, want: %v", gotStatus, wantStatus)
+	if gotStatus.Equal(wantStatus) {
```

Comment:
🙃

```suggestion
	if !gotStatus.Equal(wantStatus) {
```

Reply:
Actually, status.Equal is not good at checking equality when it comes to error status. 
So, I'm not sure if we can use status.Equal here or not. (Let's see

Reply:
>  status.Equal is not good at checking equality when it comes to error status.

This is because of this line:
https://github.com/sanposhiho/kubernetes/blob/6250971d023998b163d5381ba331b9ee14502971/pkg/scheduler/framework/interface.go#L250

(`cmp.Equal` with `cmpopts.EquateErrors()` means comparing errors via `errors.Is`.)
https://go.dev/play/p/p_bUQR8ScKJ

Reply:
Can we only compare by Message?

Reply:
Yep. If you do so, you need to check status.Code() before to make sure returned status is error.

Reply:
I just left my laptop so I'll do it later! Thanks.